x/staking/src/types/tx.rs

```diff
@@ -285,145 +305,166 @@ impl Description {
+            pubkey: Some(gears::crypto::public::PublicKey::from(msg.pub_key).into()),
+            value: Some(msg.value.into()),
         }
     }
 }
 
+impl TryFrom<inner::MsgCreateValidator> for CreateValidator {
+    type Error = CoreError;
+
+    fn try_from(val: inner::MsgCreateValidator) -> Result<Self, Self::Error> {
+        // TODO: there are missing checks here, like that validator address and delegator address should be the same
+        // see https://github.com/cosmos/cosmos-sdk/blob/2582f0aab7b2cbf66ade066fe570a4622cf0b098/x/staking/types/msg.go#L89-L144
+        let description = val
+            .description
+            .ok_or(CoreError::MissingField("description".into()))?;
+        let commission = val
+            .commission
+            .ok_or(CoreError::MissingField("commission".into()))?;
+        let pubkey = val.pubkey.ok_or(CoreError::MissingField("pubkey".into()))?;
+        let pubkey = gears::crypto::public::PublicKey::try_from(pubkey)
+            .map_err(|e| CoreError::DecodeGeneral(e.to_string()))?;
+
+        Ok(CreateValidator {
+            description: Description {
+                moniker: description.moniker,
+                identity: description.identity,
+                website: description.website,
+                security_contact: description.security_contact,
+                details: description.details,
+            },
+            commission: CommissionRates {
+                rate: Decimal256::from_cosmos_proto_string(&commission.rate)
+                    .map_err(|e| CoreError::DecodeGeneral(e.to_string()))?,
+                max_rate: Decimal256::from_cosmos_proto_string(&commission.max_rate)
+                    .map_err(|e| CoreError::DecodeGeneral(e.to_string()))?,
+                max_change_rate: Decimal256::from_cosmos_proto_string(&commission.max_change_rate)
+                    .map_err(|e| CoreError::DecodeGeneral(e.to_string()))?,
+            },
+            min_self_delegation: Uint256::from_str(&val.min_self_delegation)
+                .map_err(|e| CoreError::DecodeGeneral(e.to_string()))?,
+            delegator_address: AccAddress::from_bech32(&val.delegator_address)
+                .map_err(|e| CoreError::DecodeAddress(e.to_string()))?,
+            validator_address: ValAddress::from_bech32(&val.validator_address)
+                .map_err(|e| CoreError::DecodeAddress(e.to_string()))?,
+            pub_key: pubkey.into(),
+            value: val
+                .value
+                .ok_or(CoreError::MissingField("value".into()))?
+                .try_into()
+                .map_err(|e| CoreError::Coin(format!("{e}")))?,
+        })
+    }
+}
+
 /// CreateValidator defines a SDK message for creating a new validator.
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 pub struct CreateValidator {
     pub description: Description,
-    pub commission: CommissionRates,
+    pub commission: CommissionRates, //TODO: should name be commission_rate?
```

Comment:
If you want, but don't that you should worry about this. There type with correct name

Reply:
I wanted it to be consistent with Cosmos to avoid confusion... turns out it is so I've removed the TODO